core/server/src/main/java/alluxio/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -147,11 +137,19 @@ public FilePersister(FileDataManager fileDataManager, long fileId, List<Long> bl
 
     @Override
     public void run() {
-      try {
+      if (mFileDataManager.needPersistence(mFileId)) {
+        // lock all the blocks of the file to prevent eviction
```

Comment:
Never mind, we didn't make this guarantee before anyway. When we add the guarantee we will probably do it on the writing to memory path and it won't matter what we do here.